Description:
 Disable z3 to avoid pulling ocaml into main.
 For some reason the cmake option LLVM_ENABLE_Z3_SOLVER was not taken into account
Author: Gianfranco Costamagna <locutusofborg@debian.org>

Last-Update: 2019-11-26

Index: llvm-toolchain-snapshot_10~+20191225062409+aff6c9db9a9/llvm/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot_10~+20191225062409+aff6c9db9a9.orig/llvm/CMakeLists.txt
+++ llvm-toolchain-snapshot_10~+20191225062409+aff6c9db9a9/llvm/CMakeLists.txt
@@ -351,24 +351,22 @@ option(LLVM_ENABLE_ZLIB "Use zlib for co
 
 set(LLVM_Z3_INSTALL_DIR "" CACHE STRING "Install directory of the Z3 solver.")
 
-find_package(Z3 4.7.1)
-
-if (LLVM_Z3_INSTALL_DIR)
-  if (NOT Z3_FOUND)
-    message(FATAL_ERROR "Z3 >= 4.7.1 has not been found in LLVM_Z3_INSTALL_DIR: ${LLVM_Z3_INSTALL_DIR}.")
-  endif()
-endif()
-
-set(LLVM_ENABLE_Z3_SOLVER_DEFAULT "${Z3_FOUND}")
-
 option(LLVM_ENABLE_Z3_SOLVER
   "Enable Support for the Z3 constraint solver in LLVM."
-  ${LLVM_ENABLE_Z3_SOLVER_DEFAULT}
+  ON
 )
 
 if (LLVM_ENABLE_Z3_SOLVER)
-  if (NOT Z3_FOUND)
-    message(FATAL_ERROR "LLVM_ENABLE_Z3_SOLVER cannot be enabled when Z3 is not available.")
+  find_package(Z3 4.7.1)
+
+  if (LLVM_Z3_INSTALL_DIR)
+    if (NOT Z3_FOUND)
+      message(FATAL_ERROR "Z3 >= 4.7.1 has not been found in LLVM_Z3_INSTALL_DIR: ${LLVM_Z3_INSTALL_DIR}.")
+    endif()
+  else()
+    if (NOT Z3_FOUND)
+      message(FATAL_ERROR "LLVM_ENABLE_Z3_SOLVER cannot be enabled when Z3 is not available.")
+    endif()
   endif()
 
   set(LLVM_WITH_Z3 1)
